packages/grafana-data/src/field/fieldComparers.ts

```diff
@@ -79,9 +79,14 @@ const falsyComparer = (a: unknown, b: unknown): number => {
   return 0;
 };
 
-const timestampIndexComparer = (values: number[], reverse: boolean): IndexComparer => {
+const timestampIndexComparer = (values: number[], reverse: boolean, nanos?: number[]): IndexComparer => {
   let mult = reverse ? -1 : 1;
-  return (a: number, b: number): number => mult * (values[a] - values[b]);
+  return (a: number, b: number): number => {
+    if (values[a] === values[b] && nanos?.[a] !== undefined && nanos?.[b] !== undefined) {
+      return mult * (nanos[a] - nanos[b]);
+    }
+    return mult * (values[a] - values[b]);
+  };
```

Comment:
you should avoid checking if nanos is defined in the sorter.

instead you can have `timestampIndexComparer` return one function where nanos is not defined or used and another where it's guaranteed to be there.

something like

```js
if (nanos != null) {
   return (a: number, b: number): number => mult * (values[a] - values[b] || nanos[a] - nanos[b]);
}

return (a: number, b: number): number => mult * (values[a] - values[b]);
```